src/network/WaitStrategy.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ILogger} from '../logging/ILogger';
+import * as Promise from 'bluebird';
+
+export class WaitStrategy {
+    private readonly initialBackoffMillis: number;
+    private readonly maxBackoffMillis: number;
+    private readonly multiplier: number;
+    private readonly jitter: number;
+    private readonly clusterConnectTimeoutMillis: number;
+    private logger: ILogger;
+    private attempt: number;
+    private currentBackoffMillis: number;
+    private clusterConnectAttemptBegin: number;
+
+    constructor(initialBackoffMillis: number, maxBackoffMillis: number, multiplier: number, clusterConnectTimeoutMillis: number,
+                jitter: number, logger: ILogger) {
+        this.initialBackoffMillis = initialBackoffMillis;
+        this.maxBackoffMillis = maxBackoffMillis;
+        this.multiplier = multiplier;
+        this.clusterConnectTimeoutMillis = clusterConnectTimeoutMillis;
+        this.jitter = jitter;
+        this.logger = logger;
+    }
+
+    public reset(): void {
+        this.attempt = 0;
+        this.clusterConnectAttemptBegin = Date.now();
+        this.currentBackoffMillis = Math.min(this.maxBackoffMillis, this.initialBackoffMillis);
+    }
+
+    public sleep(): Promise<boolean> {
+        this.attempt++;
+        const currentTimeMillis = Date.now();
+        const timePassed = currentTimeMillis - this.clusterConnectAttemptBegin;
+        if (timePassed > this.clusterConnectTimeoutMillis) {
+            this.logger.warn('WaitStrategy', `Unable to get live cluster connection,
```

Comment:
nit: multiline template literals will be printed as is into the console. Not sure if that was intended here and in the next logging call.

Reply:
Thanks for that. I wanted it to be logged in a single line.